redash/models.py

```diff
@@ -658,10 +660,9 @@ def all_queries(cls, groups, drafts=False):
             .order_by(cls.created_at.desc())
 
         if drafts:
-            q = q.where(Query.name == 'New Query')
+            q = q.where(Query.is_draft == True)
         else:
-            q = q.where(Query.name != 'New Query')
-
+            q = q.where(Query.is_draft == False)
```

Comment:
👍 to adding this migration.

As for the other question - basically we want to maintain the balance between "open by default" to have some room for people to work in "private". I can think of a few options:

1. Have all queries (draft or not) discoverable via search, but drafts won't appear in "Recent" queries or the main queries list.
2. Maintain the current behavior: new query starts as a draft, but once the user gives it a name we promote it to published status. The user can choose to unpublish it.
3. Option #2 + change the query rename UI to have two buttons: "Save & Publish" (default) and "Save". This to reduce friction from the flow where you want to name a draft but not publish it.

Each options has its pros and cons. The main benefit of #2 is that it maintains current behavior & easy to implement.